protocol/blocks.go

```diff
@@ -39,5 +38,11 @@ func (m *Manager) OnNewBlock(block *util.Block) error {
 	copy(txIDsToBroadcast[len(transactionsAcceptedToMempool):], txIDsToBroadcast)
 	txIDsToBroadcast = txIDsToBroadcast[:wire.MaxInvPerTxInvMsg]
 	inv := wire.NewMsgTxInv(txIDsToBroadcast)
-	return m.netAdapter.Broadcast(peerpkg.ReadyPeerIDs(), inv)
+	return m.netAdapter.Broadcast(m.peers.ReadyPeerIDs(), inv)
+}
+
+// AddBlock adds the given block to the DAG and propagates it.
+func (m *Manager) AddBlock(block *util.Block) error {
+	// TODO(libp2p): unimplemented
```

Comment:
Please create ticket, since it's core functionality

Reply:
This is beyond "core functionality" - this+getBlockTemplate are the first and foremost rpc calls we are going to use in tests, without them there's little point to RPC.

I'd rather if you implemented this here, or at the very least - this should be the next ticket you do.

Reply:
Created.